### PR TITLE
[PLAYER-2692] Initial commit Discovery Sample App

### PR DESCRIPTION
**Pull request by ipadilla4**
https://jira.corp.ooyala.com/browse/PLAYER-2692

### CODE REVIEW

**Review by brol1dev — CHANGES_REQUESTED** [review]
This PR is hard to review. I see that you added all of the `skin-config/node_modules` folder. Remove it, you don't need it.

**Comment by DmitriiMaslov** [comment]
@ipadilla4 Hello Ileana, is it a valid PR? Do you think we should review it again and merge?
@maxkupetskii  FYI

**Comment by ipadilla4** [comment]
@DmitriiMaslov yes I think it can be useful to have this sample app on Github. Please review it.

**Comment by maxkupetskii** [comment]
@DmitriiMaslov @ipadilla4 

I had to merge this branch with the current `candidate` branch in order to test if it works. Also I made some small changes so now it matches other sample apps. Further on I'll make comments according to the updated version

**Review by maxkupetskii — CHANGES_REQUESTED** [review]
1. Fix storyboards/xibs, since they contain errors and warnings
2. Remove obsolete/deprecated methods from the project
3. Remove 3rd party dependency `CJSONSerializer` -- now it can be made with Foundation
4. Remove network code from the view controllers
5. Remove multiple casting (to`NSDictionary` for example)
6. Update all methods signatures so it won't start with capitalized letters
7. Remove network code from `cellForRowAtIndexPath`
8. Remove `decodeString` from the view controllers
9. Since `MainView` and `BrowseView` share the same code it's better to use either inheritance or move same methods (`GetDiscoveryData:::` for example) to a different class

**Review by maxkupetskii — APPROVED** [review]

File: DiscoverySampleApp/DiscoverySampleApp/BrowseView.m [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+}
+
+- (void)didUpdateAssetTitle:(NSString *)title {
+  dispatch_async(dispatch_get_main_queue(), ^{
+    self.videoTitle.text = title;
+  });
+}
+
+- (void)didStartPlaying {
+//  [self updateScrollViewWithDiscovery];
```
Comment (Artur-Sakhno) [inline DiscoverySampleApp/DiscoverySampleApp/BrowseView.m:149]: Should not we remove commented code?

File: DiscoverySampleApp/DiscoverySampleApp/Players/PlayerViewController.m [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+- (void)setCustomSkin {
+  // Setting Discovery on player
+  OODiscoveryOptions *discoveryOptions = [[OODiscoveryOptions alloc] initWithType:OODiscoveryTypePopular
+                                                                            limit:10
+                                                                          timeout:60];
+  // Modifying PlayerViewController to use Ooyala Skin
+  NSURL *jsCodeLocation = [NSBundle.mainBundle URLForResource:@"main" withExtension:@"jsbundle"];
+//  NSURL *jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.ios.bundle?platform=ios&dev=true"];
```
Comment (Artur-Sakhno) [inline DiscoverySampleApp/DiscoverySampleApp/Players/PlayerViewController.m:51]: same

**Review by Artur-Sakhno — COMMENTED** [review]
Also i have question about outlets, is it better to use strong over weak?

File: DiscoverySampleApp/DiscoverySampleApp/Players/PlayerViewController.m (hunk 2 above)
Comment (maxkupetskii) [reply DiscoverySampleApp/DiscoverySampleApp/Players/PlayerViewController.m:51]: It's an option for `main.jsbundle` to switch between a local file and a running on localhost

File: DiscoverySampleApp/DiscoverySampleApp/BrowseView.m (hunk 1 above)
Comment (maxkupetskii) [reply DiscoverySampleApp/DiscoverySampleApp/BrowseView.m:149]: No, it's an option that can be also used

**Review by Artur-Sakhno — APPROVED** [review]

**Comment by maxkupetskii** [comment]
@Artur-Sakhno 
better to have them strong

**Pull request merged**